### ISSUE TITLE
Unable to debug when runtime is in Globals section of SAM Template

### ISSUE THREAD

**Issue by awschristou** [aws/aws-toolkit-vscode#1211]
We have a [scenario](https://github.com/aws/aws-toolkit-vscode/issues/651#issuecomment-664538854) where the runtime could be specified in the Globals section instead of directly in a Resource.

#651 shows how the Globals section is not used correctly in the current (1.11.0) toolkit. I've tried it in the upcoming debug experience (#1146), and it runs into validation checks (SAM debug: invalid config: **Runtime** for Template Resource HelloWorldFunction in Template file xxx.yaml is either **undefined or unsupported**.)

(cc @justinmk3 )

**Comment by justinmk3** [comment]
This replaces/updates https://github.com/aws/aws-toolkit-vscode/issues/651 , so that can be closed after the debugconfig work is released.

**Comment by cvharris** [comment]
This only seems to be an issue for the `target: template` config. It completely ignores both the Globals section of the template.yaml as well as the `runtime` parameter of the launch config:

```json
    {
      "type": "aws-sam",
      "request": "direct-invoke",
      "name": "LambdaHandler",
      "invokeTarget": {
        "target": "template",
        "templatePath": "template.yaml",
        "logicalId": "LambdaHandler"
      },
      "lambda": {
        "runtime": "nodejs12.x",
        "payload": {}
        }
      }
    }
```

**Comment by justinmk3** [comment]
> only seems to be an issue for the `target: template` config. 

`target:code` by design ignores the SAM template (template.yaml), so I would expected it to also "ignore" Globals section :)

> It completely ignores both the Globals section of the template.yaml as well as the `runtime` parameter of the launch config:

For `target:template`, the template.yaml dictates the runtime, not the launch config. Although, I agree this is confusing, because the launch config should be the authority.  Or the Toolkit should disallow `lambda.runtime` for `target:template`.

**Comment by justinmk3** [comment]
Fixed in Toolkit 1.20
